client/deposits/instant-deposits/index.js

```diff
@@ -0,0 +1,78 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import InstantDepositModal from './modal';
+
+const InstantDepositButton = ( {
+	// eslint-disable-next-line camelcase
+	balance: { amount, fee, net, transaction_ids },
+} ) => {
+	const [ isModalOpen, setModalOpen ] = useState( false );
+	const [ inProgress, setInProgress ] = useState( false );
+	const [ hasNotice, setHasNotice ] = useState( false );
```

Comment:
The name `hasNotice` makes it sound like it's always a boolean. Instead, it can be a boolean, the string `"success"`, or an error object. Let's change the name to `notice / setNotice` and have its possible values be a bit more consistent, like for example `null / { result: 'success' } / { result: 'error', code: '...' }`.

Reply:
Makes sense, I had originally had `hasError`, then switched it to be dual purpose. I've updated it and will keep this in mind for the future.